### PR TITLE
Adding tests for `mint` and `burn`

### PR DESCRIPTION
**Pull request by nfurfaro**
This adds some tests for the stdlib functions added here: https://github.com/FuelLabs/sway-lib-std/pull/26
As usual, this build will fail until the other PR is merged to the stdlib `master` branch.

### CODE REVIEW

File: test/src/e2e_vm_tests/test_programs/test_fuel_coin_abi/Forc.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[project]
+author  = "Nick Furfaro"
+license = "MIT"
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/test_fuel_coin_abi/Forc.toml:3]:
```suggestion
author  = "Fuel Labs <contact@fuel.sh>"
license = "Apache-2.0"
```

File: test/src/e2e_vm_tests/test_programs/test_fuel_coin_abi/Forc.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+name = "test_fuel_coin_abi"
+entry = "main.sw"
+
+[dependencies]
+std = { git = "http://github.com/FuelLabs/sway-lib-std" }
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/test_fuel_coin_abi/Forc.toml:8]:
```suggestion
std = { git = "http://github.com/FuelLabs/sway-lib-std", version = "v0.0.1" }

```

File: test/src/e2e_vm_tests/test_programs/test_fuel_coin_contract/Forc.toml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[project]
+author  = "Nick Furfaro"
+license = "MIT"
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/test_fuel_coin_contract/Forc.toml:3]:
```suggestion
author  = "Fuel Labs <contact@fuel.sh>"
license = "Apache-2.0"
```

File: test/src/e2e_vm_tests/test_programs/test_fuel_coin_contract/Forc.toml [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+name = "test_fuel_coin_contract"
+entry = "main.sw"
+
+[dependencies]
+std = { git = "http://github.com/FuelLabs/sway-lib-std" }
+test_fuel_coin_abi = { path = "../test_fuel_coin_abi"}
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/test_fuel_coin_contract/Forc.toml:9]:
```suggestion
std = { git = "http://github.com/FuelLabs/sway-lib-std", version = "v0.0.1" }
test_fuel_coin_abi = { path = "../test_fuel_coin_abi" }

```

File: test/src/e2e_vm_tests/test_programs/token_ops_test/Forc.toml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+[project]
+author  = "Nick Furfaro"
+license = "MIT"
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/token_ops_test/Forc.toml:3]:
```suggestion
author  = "Fuel Labs <contact@fuel.sh>"
license = "Apache-2.0"
```

File: test/src/e2e_vm_tests/test_programs/token_ops_test/Forc.toml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+name = "token_ops_test"
+entry = "main.sw"
+
+[dependencies]
+std = { git = "http://github.com/FuelLabs/sway-lib-std" }
+core = { git = "http://github.com/FuelLabs/sway-lib-core" }
+test_fuel_coin_abi = { path = "../test_fuel_coin_abi"}
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/token_ops_test/Forc.toml:10]:
```suggestion
std = { git = "http://github.com/FuelLabs/sway-lib-std", version = "v0.0.1" }
core = { git = "http://github.com/FuelLabs/sway-lib-core", version = "v0.0.1" }
test_fuel_coin_abi = { path = "../test_fuel_coin_abi" }
```

File: test/src/e2e_vm_tests/test_programs/token_ops_test/Forc.toml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+# test_fuel_coin_contract
+[[tx-input]]
+type = "Contract"
+contract-id = "0x16af825dc6a612810b6c80a16f2a7e3248f574f2b0c96babda736fc53d8d8f8c"
+utxo-id = "0xad6aaaa1d6fd78f91693ee2cc124fd43d25bd1c015b88b675ee43d6b5e14058600"
+balance-root = "0xad6aaaa1d6fd78f91693ee2cc124fd43d25bd1c015b88b675ee43d6b5e140586"
+state-root = "0xad6aaaa1d6fd78f91693ee2cc124fd43d25bd1c015b88b675ee43d6b5e140586"
+
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/token_ops_test/Forc.toml:19]: Hmmm. Why does including this not error? It should have been removed in #675. cc @sezna, does including sections that aren't known not result in a error?
```suggestion
```

Comment (sezna) [reply test/src/e2e_vm_tests/test_programs/token_ops_test/Forc.toml:19]: Yeah, just tested and it looks like it ignores unknown stuff. I know rust allows it, too, but provides a warning like "this manifest key doesn't do anything."

Comment (adlerjohn) [reply test/src/e2e_vm_tests/test_programs/token_ops_test/Forc.toml:19]: Filed https://github.com/FuelLabs/sway/issues/694

**Review by adlerjohn — APPROVED** [review]
LGTM, will defer to @sezna

**Review by sezna — APPROVED** [review]

**Pull request merged**